evaluation.py

```diff
@@ -146,6 +147,101 @@ def _non_max_supression(self, plain, threshold, windowSize=3):
         under_thresh_indices = plain < threshold
         plain[under_thresh_indices] = 0
         return plain * (plain == maximum_filter(plain, footprint=np.ones((windowSize, windowSize))))
+        img_v_resize = self._vstack_images(heatmap_imgs)
+
+        cv2.imwrite(filename, img_v_resize)
+
+    """
+        Parameters
+        ----------
+        metadata : object
+        should be metadata associated to a single image
+
+        untransformed_x : int
+        x coordinate to
+    """
+    def __undo_x(self, metadata, untransformed_x):
+      predicted_x = round(untransformed_x * metadata['cropped_width'] / metadata['input_dim'][0] + metadata['anchor_x'])
+      return int(predicted_x)
+
+    """
+        Parameters
+        ----------
+        metadata : object
+        should be metadata associated to a single image
+
+        untransformed_y : int
+        x coordinate to
+    """
+    def __undo_y(self, metadata, untransformed_y):
+      predicted_y = round(untransformed_y * metadata['cropped_height'] / metadata['input_dim'][1] + metadata['anchor_y'])
+      return int(predicted_y)
+
+    """
+        Parameters
+        ----------
+        metadata : object
+        should be metadata associated to a single image
+
+        untransformed_predictions : list
+        a list of precitions that need to be transformed
+        Example:  [1,2,0,1,4,666,32...]
+    """
+    def undo_bounding_box_transformations(self, metadata, untransformed_predictions):
+      untransformed_predictions = np.array(untransformed_predictions).flatten()
+      predicted_labels = []
+      list_of_scores = []
+      for i in range(len(untransformed_predictions)):
+        if( i % 3 == 0 ): # is an x-coord
+          predicted_labels.append(self.__undo_x(metadata, untransformed_predictions[i]))
+        elif( i % 3 == 1 ): # is a y-coord
+          predicted_labels.append(self.__undo_y(metadata, untransformed_predictions[i]))
+        elif( i % 3 == 2) : # is v (visbility)
```

Comment:
nit: we could iterate from `i in range(NUM_COCO_KP)` here, and then get x,y,score from `i, i+1, i+2`. I think it would be a bit cleaner

Reply:
The if condition here is based on values of i and not values of `untransformed_predictions[i]` so I don't think we should face that issue as long as our input is always the same size. I do agree that it could be cleaned up in a future refactor, I am hesitant to refactor this now as it would require testing that I cannot do at the moment.